cli/cmd/docs/generate.go

```diff
@@ -0,0 +1,193 @@
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func GenerateCmd(rootCmd *cobra.Command) *cobra.Command {
+	docsCmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate CLI documentation",
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			dir := "docs/docs/references/CLI/"
+			if len(args) > 0 {
+				dir = args[0]
+			}
+			rootCmd.DisableAutoGenTag = true
+			err := GenMarkdownTree(rootCmd, dir)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	return docsCmd
+}
+
+func GenMarkdownTree(cmd *cobra.Command, dir string) error {
+	identity := func(s string) string {
+		parts := strings.Split(s, "_")
+		last := &parts[len(parts)-1]
+		*last = (*last)[:len(*last)-3]
+		return filepath.Join(parts...)
+	}
+	emptyStr := func(s string) string { return "" }
+	return GenMarkdownTreeCustom(cmd, dir, emptyStr, identity)
+}
+
+// GenMarkdownTreeCustom is the the same as GenMarkdownTree, but
+// with custom filePrepender and linkHandler.
+func GenMarkdownTreeCustom(cmd *cobra.Command, dir string, filePrepender, linkHandler func(string) string) error {
+	if cmd.Hidden {
+		return nil
+	}
+
+	for _, c := range cmd.Commands() {
+		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		sd := filepath.Join(dir, cmd.Name())
+		if _, err := os.Stat(sd); os.IsNotExist(err) {
+			if err := os.Mkdir(sd, fs.ModePerm); err != nil {
+				return err
+			}
+		}
+		if err := GenMarkdownTreeCustom(c, sd, filePrepender, linkHandler); err != nil {
+			return err
+		}
+	}
+
+	nm := cmd.Name()
+	filename := filepath.Join(dir, nm+".md")
+	if len(cmd.Commands()) > 0 {
+		filename = filepath.Join(dir, cmd.Name(), nm+".md")
+	}
+
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	if _, err := f.WriteString(filePrepender(filename)); err != nil {
+		return err
+	}
+
+	return GenMarkdownCustom(cmd, f, linkHandler)
+}
+
+func hasSeeAlso(cmd *cobra.Command) bool {
+	if cmd.HasParent() {
+		return true
+	}
+	for _, c := range cmd.Commands() {
+		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+func printOptions(buf *bytes.Buffer, cmd *cobra.Command, name string) error {
+	flags := cmd.NonInheritedFlags()
+	flags.SetOutput(buf)
+	if flags.HasAvailableFlags() {
+		buf.WriteString("### Flags\n\n```\n")
+		flags.PrintDefaults()
+		buf.WriteString("```\n\n")
+	}
+
+	parentFlags := cmd.InheritedFlags()
+	parentFlags.SetOutput(buf)
+	if parentFlags.HasAvailableFlags() {
+		buf.WriteString("### Global flags\n\n```\n")
+		parentFlags.PrintDefaults()
+		buf.WriteString("```\n\n")
+	}
+	return nil
+}
+
+type byName []*cobra.Command
+
+func (s byName) Len() int           { return len(s) }
+func (s byName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s byName) Less(i, j int) bool { return s[i].Name() < s[j].Name() }
```

Comment:
Move helpers to after `GenMarkdownCustom` (so functions roughly follow call order)